tempfile_linux.go

```diff
@@ -0,0 +1,32 @@
+package extsort
+
+import "os"
+
+func newTempFile(dir, pattern string, keepFile bool) (*os.File, error) {
+	f, err := os.CreateTemp(dir, pattern)
+	if err != nil {
+		return nil, err
+	}
+	if keepFile {
+		return f, nil
+	}
+	// immediately remove for less chance of orphaning.
+	if err := os.Remove(f.Name()); err != nil {
```

Comment:
Sorry to ask, but what's the purpose of this? All temp files will/should be removed once sorting is completed, so why would we remove them before they are written?

Reply:
If a process crashes for whatever reason but the system doesn't restart for a long time (thus not clearing temp) you can build up a lot of orphan files, potentially wasting space or running out. Removing here as soon as it is created, means if the process goes away the file will be removed by the OS. The gap between creation and here is very small compared to the gap between creation and the end of sorting. This is the "old school" way for O_TMPFILE functionality before that was available.